src/main/java/com/dabsquared/gitlabjenkins/connection/GitLabConnectionConfig.java

```diff
@@ -92,99 +66,6 @@ public GitLabClient getClient(String connectionName, Item item, String jobCreden
         return connectionMap.get(connectionName).getClient(item, jobCredentialId);
     }
 
-    public FormValidation doCheckName(@QueryParameter String id, @QueryParameter String value) {
-        if (StringUtils.isEmptyOrNull(value)) {
-            return FormValidation.error(Messages.name_required());
-        } else if (connectionMap.containsKey(value) && !connectionMap.get(value).toString().equals(id)) {
```

Comment:
This check is very hard to do, even prior to my changes this will only work when saving.
Not even sure why a transient map is used for lookups.